src/Microsoft.ML.Transforms/PermutationFeatureImportanceExtensions.cs

```diff
@@ -348,6 +648,94 @@ private static double[] ComputeSequenceDeltas(IReadOnlyList<double> a, IReadOnly
             return delta;
         }
 
+        private static ImmutableDictionary<string, TResult>
+            PermutationFeatureImportance<TMetric, TResult>(
+                IHostEnvironment env,
+                ITransformer model,
+                IDataView data,
+                Func<TResult> resultInitializer,
+                Func<IDataView, TMetric> evaluationFunc,
+                Func<TMetric, TMetric, TMetric> deltaFunc,
+                int permutationCount,
+                bool useFeatureWeightFilter,
+                int? numberOfExamplesToUse) where TResult : IMetricsStatistics<TMetric>
+        {
+            env.CheckValue(data, nameof(data));
+            env.CheckValue(model, nameof(model));
+
+            ISingleFeaturePredictionTransformer lastTransformer = null;
+
+            if (model is TransformerChain<ITransformer> chain)
+            {
+                foreach (var transformer in chain.Reverse())
+                {
+                    if (transformer is ISingleFeaturePredictionTransformer)
+                    {
+                        lastTransformer = transformer as ISingleFeaturePredictionTransformer;
```

Comment:
(nit) you shouldn't have to cast twice here. Maybe something like:

```C#
                    if (transformer is ISingleFeaturePredictionTransformer singlePredictionTransformer)
                    {
                        lastTransformer = singlePredictionTransformer;
                        break;
                    }

Reply:
Fixed.